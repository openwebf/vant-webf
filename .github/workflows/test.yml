name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
