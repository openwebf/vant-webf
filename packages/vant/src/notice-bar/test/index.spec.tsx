import { NoticeBar } from '..';
import { mount, later } from '../../../test';
import type { ComponentInstance } from '../../utils';

test('should emit close event when close icon is clicked', () => {
  const wrapper = mount(NoticeBar, {
    props: {
      mode: 'closeable',
    },
  });
  const close = wrapper.find('.van-notice-bar__right-icon');

  close.trigger('click');
  expect(wrapper.emitted<[Event]>('close')![0][0]).toBeTruthy();
});

test('should render icon slot correct', () => {
  const wrapper = mount({
    render: () => (
      <NoticeBar
        v-slots={{
          'left-icon': () => 'Custom Left Icon',
          'right-icon': () => 'Custom Right Icon',
        }}
      >
        Content
      </NoticeBar>
    ),
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should emit replay event after replay', async () => {
  const wrapper = mount(NoticeBar, {
    props: {
      text: 'foo',
    },
  });

  wrapper.find('.van-notice-bar__content').trigger('transitionend');
  await later(80);
  expect(wrapper.emitted('replay')).toBeTruthy();
});

test('should start scrolling when content width > wrap width ', async () => {
  const wrapper = mount(NoticeBar, {
    props: {
      text: 'foo',
      delay: 0,
    },
  });

  const wrap = wrapper.find('.van-notice-bar__wrap');
  const content = wrapper.find('.van-notice-bar__content');

  wrap.element.getBoundingClientRect = () =>
    ({
      width: 50,
    } as DOMRect);
  content.element.getBoundingClientRect = () =>
    ({
      width: 100,
    } as DOMRect);

  await later(50);

  expect(wrapper.html()).toMatchSnapshot();
});

test('should not start scrolling when content width > wrap width ', async () => {
  const wrapper = mount(NoticeBar, {
    props: {
      text: 'foo',
      delay: 0,
    },
  });

  const wrap = wrapper.find('.van-notice-bar__wrap');
  const content = wrapper.find('.van-notice-bar__content');

  wrap.element.getBoundingClientRect = () =>
    ({
      width: 200,
    } as DOMRect);
  content.element.getBoundingClientRect = () =>
    ({
      width: 100,
    } as DOMRect);

  await later(50);

  expect(wrapper.html()).toMatchSnapshot();
});

test('should expose reset methods', async () => {
  const wrapper = mount(NoticeBar, {
    props: {
      text: 'foo',
    },
  });

  expect((wrapper.vm as ComponentInstance).reset).toBeTruthy();
});
