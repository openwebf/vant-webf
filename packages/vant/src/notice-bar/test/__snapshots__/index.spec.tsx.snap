// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not start scrolling when content width > wrap width  1`] = `
<div role="alert"
     class="van-notice-bar"
>
  <div role="marquee"
       class="van-notice-bar__wrap"
  >
    <div style="transition-duration: 0s;"
         class="van-notice-bar__content"
    >
      foo
    </div>
  </div>
</div>
`;

exports[`should render icon slot correct 1`] = `
<div role="alert"
     class="van-notice-bar"
>
  Custom Left Icon
  <div role="marquee"
       class="van-notice-bar__wrap"
  >
    <div style="transition-duration: 0s;"
         class="van-notice-bar__content"
    >
      Content
    </div>
  </div>
  Custom Right Icon
</div>
`;

exports[`should start scrolling when content width > wrap width  1`] = `
<div role="alert"
     class="van-notice-bar"
>
  <div role="marquee"
       class="van-notice-bar__wrap"
  >
    <div style="transition-duration: 1.6666666666666667s; transform: translateX(-100px);"
         class="van-notice-bar__content"
    >
      foo
    </div>
  </div>
</div>
`;
